components/base/kinematicbase/execution.go

```diff
@@ -415,6 +433,19 @@ func (ptgk *ptgBaseKinematics) courseCorrect(
 			// Now, the new start configuration should be [-160, -160] and the new end configuration should be [0, -160]. RSDK-7515 will
 			// simplify this significantly.
 			startVal := connectionPoint.subTraj[solution.trajIdx].Dist
+
+			// We need to know the point along the segment where we are expecting to reconnect
+			skippedSegment := []referenceframe.Input{
+				connectionPoint.arcSegment.EndConfiguration[ptgIndex],
+				connectionPoint.arcSegment.EndConfiguration[trajectoryAlphaWithinPTG],
+				connectionPoint.arcSegment.EndConfiguration[startDistanceAlongTrajectoryIndex],
+				{startVal},
+			}
+			skippedPose, err := ptgk.Kinematics().Transform(skippedSegment)
```

Comment:
This is a bug being fixed. The start pose of the newly connected segment was not quite correct before; it was the point solved for by course correction, rather than the point on the plan. It should be the point on the plan, so that inexact course corrections do not propagate error.